pages/api/upload/index.js

```diff
@@ -0,0 +1,42 @@
+import formidable from "formidable";
+import cloudinary from "cloudinary";
+
+cloudinary.config({
+  cloud_name: process.env.CLOUDINARY_CLOUD_NAME,
+  api_key: process.env.CLOUDINARY_API_KEY,
+  api_secret: process.env.CLOUDINARY_SECRET,
+});
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+export default async function handler(request, response) {
+  if (request.method !== "POST") {
+    return response.status(400).json({ message: "Method not allowed" });
+  }
+
+  const form = formidable({});
+
+  const [fields, files] = await form.parse(request);
+
+  const file = files.image[0];
+  const { newFilename, filepath } = file;
+
+  const {
+    height,
+    width,
+    secure_url: url,
+  } = await cloudinary.v2.uploader.upload(filepath, {
+    public_id: newFilename,
+    folder: "plants",
+  });
+
+  response.status(201).json({
+    height,
+    width,
+    url,
+  });
```

Comment:
Not requiring any change here... but we might want to think about what we do if something goes wrong. i.e. what if the cloudinary api is down, or our credentials are no longer valid. 

We could consider a `try catch` for this, where if something goes wrong then we log it on the server and send a message back to the client indicating that there has been an error. Then on the client side we could indicate to the user that something went wrong.

Reply:
Great idea, now everything works when Cloudinary is down. Thanks a lot for your review!